typegen/strats/inline.py

```diff
@@ -324,14 +350,66 @@ def leave_Assign(
             value=original_node.value,
         )
 
+    def leave_AnnAssign(
+        self, original_node: cst.AnnAssign, updated_node: cst.AnnAssign
+    ) -> cst.Assign | cst.AnnAssign | cst.RemovalSentinel:
+        local_var, class_member, targets = self._get_trace_for_targets(original_node)
+
+        # only one target is possible
+        tgt_cnt = len(targets.attrs) + len(targets.names)
+        assert tgt_cnt == 1, f"Only exactly one target is possible, found {tgt_cnt}"
+
+        ident, ident_node = next(itertools.chain(targets.attrs, targets.names))
+        logger.debug(f"Searching for hints to '{ident}' for an Assign")
+
+        hinted = local_var if isinstance(ident_node, cst.Name) else class_member
+        assert hinted.shape[0] <= 1, f"Found more than one type hint for '{ident}'"
+
+        if hinted.empty and original_node.value is None:
+            logger.debug(
+                "Removing AnnAssign without value \
+                    because no type hint can be provided"
+            )
+            return cst.RemoveFromParent()
+
+        elif hinted.empty and original_node.value is not None:
+            logger.debug(
+                "Replacing AnnAssign with value by Assign without type hint \
+                    because no type hint can be provided"
+            )
+            return cst.Assign(
+                targets=[cst.AssignTarget(original_node.target)],
+                value=original_node.value,
+            )
+
+        else:
+            hint_ty = hinted[TraceData.VARTYPE].values[0]
+            assert hint_ty is not None
+
+            logger.debug(f"Using '{hint_ty}' for the AnnAssign with '{ident}'")
+
+            # Replace simple assignment with annotated assignment
+            return updated_node.with_changes(
+                target=original_node.target,
+                annotation=cst.Annotation(cst.Name(value=hint_ty)),
+                value=original_node.value,
+            )
+
```

Comment:
It is quite hard to tell which parts of the code add type hints, or remove type hints. If they are implemented in different functions with no overlap, add region or a comment separating the functionalities. If they are mixed within functions, please split their implementations (Add_annotation_...() & Remove_annotation_...()).